app/src/main/java/com/github/onlynotesswent/utils/Course.kt

```diff
@@ -0,0 +1,87 @@
+package com.github.onlynotesswent.utils
+
+/**
+ * Represents a course that a note belongs to.
+ *
+ * @param courseCode The code of the course.
+ * @param courseName The name of the course.
+ * @param courseYear The year of the course.
+ * @param publicPath The public path of the course.
+ */
+data class Course(
+    val courseCode: String,
+    val courseName: String,
+    val courseYear: Int,
+    val publicPath: String
+) {
+  /**
+   * Returns the full name of the course, which is the concatenation of the course code and name.
+   *
+   * @return The full name of the course.
+   */
+  fun fullName() = "$courseCode: $courseName"
+
+  companion object {
+    // default course
+    val DEFAULT = Course("CS-311", "The Software Enterprise", 2024, "default")
+    // course code max length
+    private const val COURSE_CODE_MAX_LENGTH = 10
+    // course name max length
+    private const val COURSE_NAME_MAX_LENGTH = 50
+
+    private val LOWERCASE_WORDS =
```

Comment:
Why are you using a list of word instead of just capitalizing the first letter of every word using the whitespaces ?

Reply:
I tried implementing Title Case by adding a list of small words that won't get capitalized. This way it will look like:

BIO-333: Biology of the Cell

Instead of:

BIO-333: Biology Of The Cell

If you don't think Title Case is worth it and capitalizing each word is better then we can change it now. 

But we can always change this in the future very easily.

Reply:
Ok cool i like it